mithril-aggregator/src/artifact_builder/cardano_database_artifacts/ancillary.rs

```diff
@@ -48,21 +145,43 @@ impl AncillaryArtifactBuilder {
 
 #[cfg(test)]
 mod tests {
+    use std::fs::File;
+
+    use flate2::read::GzDecoder;
     use mockall::predicate::eq;
+    use tar::Archive;
+
+    use mithril_common::digesters::{
+        DummyCardanoDbBuilder, IMMUTABLE_DIR, LEDGER_DIR, VOLATILE_DIR,
+    };
+
+    use crate::{
+        test_tools::TestLogger, CompressedArchiveSnapshotter, DumbSnapshotter,
+        SnapshotterCompressionAlgorithm,
+    };
 
     use super::*;
 
     #[tokio::test]
-    async fn upload_archive_should_return_empty_locations_with_no_uploader() {
-        let builder = AncillaryArtifactBuilder::new(vec![]);
+    async fn upload_ancillary_archive_should_return_empty_locations_with_no_uploader() {
```

Comment:
This case should trigger an error somewhere: here or in the caller as we don't want to create snasphots without ancillary locations.